internal/graph/cached_resolver_test.go

```diff
@@ -590,13 +590,13 @@ func TestCheckCacheKey_ContextualTuplesOrdering(t *testing.T) {
 	modelID := ulid.Make().String()
 
 	tuples1 := []*openfgav1.TupleKey{
-		tuple.NewTupleKey("document:1", "viewer", "user:jon"),
-		tuple.NewTupleKey("document:2", "viewer", "user:jon"),
+		tuple.NewTupleKey("document:1", "viewer", "user:anne"),
+		tuple.NewTupleKey("document:2", "admin", "user:jon"),
```

Comment:
Existing unit test only tested the sorting order of the object property but did not cover when the relation and user are differing values.

Reply:
I'm realizing now there's another test gap in our cache key computation. Can we also add a test or two asserting alternative order of tuples with conditions are also ok?

For example,

```
document:1#viewer@user:anne, condition: "mycond"
document:2#admin@user:jon

vs

document:2#admin@user:jon
document:1#viewer@user:anne, condition: "mycond"
```

We have tests that ensure two different check requests with the same context are the same, but not necessarily tests with tuples involving a condition like the one above.

Reply:
@jon-whit AFAICT, the cache key generation does not account for condition. Which explains the absence of testing but begs the question if it should be accounted for. If so, I can look into adding this capability.

Added the requested test for this anyway.

Reply:
I’ll need to think about this one more tomorrow.

You can’t have the same base triplet (e.g. `(object, relation, user)`) in the database at the same time where one is conditioned and the other is not. In other words, you can’t have

```
- document:1#viewer@user:will
- document:1#viewer@user:will, condition: "somecondition"
```
at the same time in the same store.

If we cache a Check subproblem with a tuple that is not conditioned, then no matter what you do to the tuples (deletions etc) the subproblem will be cached for the TTL regardless if you delete it and re-introduce it with a condition. So I don’t think omitting the condition from the cache key is an issue per-se, but I need to think more about it tomorrow 🤔

Reply:
If there is something actionable, can that be done in a follow-up? AFAICT, the spirit of this PR has been fulfilled and its ready to go.